### ISSUE TITLE
Support `MLFlowLogger` with only the `mlflow-skinny` dependency installed

### ISSUE THREAD

**Issue by dconathan** [Lightning-AI/lightning#16486]
### Bug description

Sorry not sure to label this as a "feature" or a "bug":

`mlflow-skinny` is a variant of the `mlflow` library that installs the bare-minimum required for tracking: https://pypi.org/project/mlflow-skinny/

As far as I can tell, this meets the requirements for using the `MLFlowLogger`, but currently it doesn't work because:

https://github.com/Lightning-AI/lightning/blob/4a802e00a843264afd5aea023d8475d4f1b4d360/src/pytorch_lightning/loggers/mlflow.py#L40

checks that precisely the `mlflow` library is installed.  If this line were replaced with:

```python
_MLFLOW_AVAILABLE = RequirementCache("mlflow>=1.0.0") or RequirementCache("mlflow-skinny>=1.0.0")
```

It seems to work fine.

Should I make a PR for this?  Would we want to add separate tests for `mlflow` and `mlflow-skinny` environments?

cc @borda

**Comment by carmocca** [comment]
Sounds good to me. You will probably need separate variables for each check in case some features are not supported with the skinny version

**Comment by adeandrade** [comment]
Why was this merged on 2.0 if it was broken on 1.9?

**Comment by carmocca** [comment]
@adeandrade We only include bug-fixes in the 1.9.x releases. And this wasn't considered one because it's adding support for this alternative MLFlow package. Does our Logger not work with `mlflow` installed?

**Comment by adeandrade** [comment]
pytorch-lightning was working with mlflow-skinny on 1.8 and it broke on 1.9 but I guess mlflow-skinny was not considered before. It's OK. I'll stick to 1.8 until 2.0 is released.
